### PR TITLE
Transaction

### PR DESCRIPTION
**Pull request by SovaV**

### CODE REVIEW

File: controllers/transaction/addTransaction.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    );
+    if (lastTransaction) {
+      type
+        ? (total = lastTransaction.total - sum)
+        : (total = lastTransaction.total + sum);
+    } else {
+      type ? (total = sum) : new Conflict();
+    }
+    if (total < 0) {
+      throw new Unauthorized();
```
Comment (Prykhodko-Pasha) [inline controllers/transaction/addTransaction.js]: This is not an Unauthorized error. It must be the same error as in the row 37

File: models/transactions.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
+const { Schema, model } = require("mongoose");
+const Joi = require("joi");
+
+const transactionSchema = Schema(
+  {
+    type: {
+      type: Boolean,
+      default: true,
```
Comment (Prykhodko-Pasha) [inline models/transactions.js]: The Type field must not has a default value. It must be a required field

File: models/transactions.js [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    sum: {
+      type: Number,
+      required: true,
+    },
+    comment: {
+      type: String,
+    },
+    total: {
+      type: Number,
+      required: true,
```
Comment (Prykhodko-Pasha) [inline models/transactions.js]: This is not required field

File: models/transactions.js [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+        "other",
+      ],
+      default: "basic",
+    },
+  },
+  { versionKey: false, timestamps: true }
+);
+
+const joiShema = Joi.object({
+  type: Joi.bool,
```
Comment (Prykhodko-Pasha) [inline models/transactions.js]: This is required field

File: package.json [hunk 5]
```diff
@@ -16,6 +16,7 @@
     "cross-env": "7.0.3",
     "dotenv": "10.0.0",
     "express": "4.17.1",
+    "http-errors": "^2.0.0",
```
Comment (Prykhodko-Pasha) [inline package.json]: Please delete the caret

File: routes/api/transactions/transactions.js [hunk 6]
```diff
@@ -0,0 +1,7 @@
+const express = require("express");
+const router = express.Router();
+const transactionController = require("../../../controllers/transaction");
```
Comment (Prykhodko-Pasha) [inline routes/api/transactions/transactions.js]: Please move this file to the folder "api", where the other route files are

File: controllers/transaction/addTransaction.js [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+    }
+    const newTransactions = await Transaction.create({
+      ...req.body,
+      sum,
```
Comment (Prykhodko-Pasha) [inline controllers/transaction/addTransaction.js]: Req.body is already contain the sum

File: controllers/transaction/addTransaction.js [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+      total,
+      // owner: _id,
+    });
```
Comment (Prykhodko-Pasha) [inline controllers/transaction/addTransaction.js:43]: Also you have to add fields month and year

File: controllers/transaction/addTransaction.js (hunk 1 above)
Comment (SovaV) [reply controllers/transaction/addTransaction.js]: +

File: models/transactions.js (hunk 2 above)
Comment (SovaV) [reply models/transactions.js]: corrected, thank you

File: models/transactions.js (hunk 3 above)
Comment (SovaV) [reply models/transactions.js]: corrected, thank you

File: models/transactions.js (hunk 4 above)
Comment (SovaV) [reply models/transactions.js]: corrected, thank you

File: package.json (hunk 5 above)
Comment (SovaV) [reply package.json]: corrected, thank you
it was a novelty))

File: routes/api/transactions/transactions.js (hunk 6 above)
Comment (SovaV) [reply routes/api/transactions/transactions.js]: corrected, thank you

File: controllers/transaction/addTransaction.js (hunk 7 above)
Comment (SovaV) [reply controllers/transaction/addTransaction.js]: corrected, thank you

File: controllers/transaction/addTransaction.js (hunk 8 above)
Comment (SovaV) [reply controllers/transaction/addTransaction.js:43]: added month and year

**Review by Prykhodko-Pasha — APPROVED** [review]

**Review by Vladomira — APPROVED** [review]

**Pull request merged**